injected/src/features/message-bridge.js

```diff
@@ -0,0 +1,168 @@
+    proxies = new Map();
+    subscriptions = new Map();
+
+    init(args) {
+        console.log(`[isolated] will install bridge with ${args.sessionKey}`);
+        /**
+         * TODO: what is the value of this token? - since 'createPageWorldBridge' is available to all page scripts, probably none
+         * @param {string} eventName
+         * @return {`${string}-${string}`}
+         */
+        function appendToken(eventName) {
+            return `${eventName}-${args.sessionKey}`;
+        }
+        window.addEventListener(appendToken(InstallProxy.NAME), (/** @type {CustomEvent<unknown>} */ e) => {
+            console.log(`[isolated:incoming] ${InstallProxy.NAME}`, e.detail);
+            const incoming = InstallProxy.create(e.detail);
+            if (!incoming) return;
+            this.installProxyFor(incoming.featureName, args.messagingConfig);
+        });
+        window.addEventListener(appendToken(ProxyNotification.NAME), (/** @type {CustomEvent<unknown>} */ e) => {
+            console.log(`[isolated:incoming] ${ProxyNotification.NAME}`, e.detail);
+            const notification = ProxyNotification.create(e.detail);
+            if (!notification) return;
+            this.proxyNotification(notification);
+        });
+        window.addEventListener(appendToken(ProxyRequest.NAME), (/** @type {CustomEvent<unknown>} */ e) => {
+            console.log(`[isolated:incoming] ${ProxyRequest.NAME}`, e.detail);
+            const request = ProxyRequest.create(e.detail);
+            if (!request) return;
+            this.proxyRequest(request, appendToken);
+        });
+        window.addEventListener(appendToken(SubscriptionRequest.NAME), (/** @type {CustomEvent<unknown>} */ e) => {
+            console.log(`[isolated:incoming] ${SubscriptionRequest.NAME}`, e.detail);
+            const subscription = SubscriptionRequest.create(e.detail);
+            if (!subscription) return;
+            this.proxySubscription(subscription, appendToken);
+        });
+        window.addEventListener(appendToken(SubscriptionUnsubscribe.NAME), (/** @type {CustomEvent<unknown>} */ e) => {
+            console.log(`[isolated:incoming] ${SubscriptionUnsubscribe.NAME}`, e.detail);
+            const unsubscribe = SubscriptionUnsubscribe.create(e.detail);
+            if (!unsubscribe) return;
+            this.removeSubscription(unsubscribe);
+        });
+    }
+
+    /**
+     * @param {string} featureName
+     * @param {import('@duckduckgo/messaging').MessagingConfig} config
+     */
+    installProxyFor(featureName, config) {
+        if (this.proxies.has(featureName)) return;
+        const allowed = this.getFeatureSetting('allowed');
+        if (!allowed.includes(featureName)) return;
+
+        const ctx = { ...this.messaging.messagingContext, featureName };
+        const messaging = new Messaging(ctx, config);
+        this.proxies.set(featureName, messaging);
+        console.log('[isolated] did install proxy for ', featureName);
```

Comment:
This is the magic - it created a new `Messaging` instance in the fly, but crucially it re-uses the per-platform config. This is very important since each platform may have varying levels of global side-effects.